primitives/src/v2/mod.rs

```diff
@@ -1569,6 +1598,72 @@ impl CompactStatement {
 	}
 }
 
+/// `IndexedVec` struct indexed by type specific indices.
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(PartialEq, MallocSizeOf))]
+pub struct IndexedVec<K, V>(Vec<V>, PhantomData<fn(K) -> K>);
```

Comment:
```suggestion
pub struct IndexedVec<K, V>(Vec<V>, PhantomData<K>);
```

Reply:
https://docs.rs/typed-index-collections/3.1.0/src/typed_index_collections/vec.rs.html#89-109

Reply:
Okay, you send me down a rabbit hole :P Do you know why `K` as invariant is not appearing in the auto trait derives? Just some Rustc magic of ignoring?

Reply:
Sorry :) 
I think auto trait derives do not suffer some regular dumb `derive(Trait)` limitations that every type mentioned must `Trait`.
In this case `fn(K) -> K` impls `Send` even if `K` doesn't: https://doc.rust-lang.org/core/primitive.fn.html#synthetic-implementations. I don't know where is this is defined though. But doesn't seem to be in
https://github.com/rust-lang/rust/blob/master/compiler/rustc_trait_selection/src/traits/auto_trait.rs

Reply:
Hmm. So does `fn(K)`. I'm not quire sure what is the point of `fn(K) -> K` then.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9510d8875227f41600b6c871f7394f91

Reply:
Haha. No problem. Seems like something interested where you can waste a lot of time on 🤣

Reply:
Tbh, I just used the code directly as I saw it... Considered `PhantomData<K>` but didn't want to have any weird issues while testing the generics.